src/main/java/io/split/android/client/storage/attributes/SqLitePersistentAttributesStorage.java

```diff
@@ -0,0 +1,65 @@
+package io.split.android.client.storage.attributes;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.split.android.client.storage.db.attributes.AttributesDao;
+import io.split.android.client.storage.db.attributes.AttributesEntity;
+import io.split.android.client.utils.Json;
+import io.split.android.client.utils.Logger;
+
+public class SqLitePersistentAttributesStorage implements PersistentAttributesStorage {
+
+    private static final Type ATTRIBUTES_MAP_TYPE = new TypeToken<Map<String, Object>>() {
+    }.getType();
+    private final AttributesDao mAttributesDao;
+    private final String mUserKey;
+
+    public SqLitePersistentAttributesStorage(@NonNull AttributesDao attributesDao, @NonNull String userKey) {
+        mAttributesDao = attributesDao;
```

Comment:
Please add checkNotNull call for @NonNull arguments
```suggestion
        mAttributesDao = checkNotNull(attributesDao);
        mUserKey = checkNotNull(userKey);
```